include/flox/search/params.hh

```diff
@@ -106,44 +108,94 @@ to_json( nlohmann::json & jto, const SearchQuery & qry );
 
 /* -------------------------------------------------------------------------- */
 
-/**
- * @brief SearchParams used to search for packages in a collection of inputs.
- *
- * Example Parameters:
- * ```
- * {
- *   "registry": {
- *     "inputs": {
- *       "nixpkgs": {
- *         "from": {
- *           "type": "github"
- *         , "owner": "NixOS"
- *         , "repo": "nixpkgs"
- *         }
- *       , "subtrees": ["legacyPackages"]
- *       }
- *     , "floco": {
- *         "from": {
- *           "type": "github"
- *         , "owner": "aakropotkin"
- *         , "repo": "floco"
- *         }
- *       , "subtrees": ["packages"]
- *       }
- *     }
- *   , "defaults": {
- *       "subtrees": null
- *     }
- *   , "priority": ["nixpkgs", "floco"]
- *   }
- * , "systems": ["x86_64-linux"]
- * , "allow":   { "unfree": true, "broken": false, "licenses": ["MIT"] }
- * , "semver":  { "preferPreReleases": false }
- * , "query":   { "match": "hello" }
- * }
- * ```
- */
-using SearchParams = pkgdb::QueryParams<SearchQuery>;
+struct SearchParams
+{
+
+  /**
+   * @brief The absolute @a std::filesystem::path to a manifest file or an
+   * inline @a flox::resolver::GlobalManifestRaw.
+   */
+  std::optional<
+    std::variant<std::filesystem::path, resolver::GlobalManifestRaw>>
+    globalManifest;
+
+  /**
+   * @brief The absolute @a std::filesystem::path to a manifest file or an
+   * inline @a flox::resolver::ManifestRaw.
+   */
+  std::variant<std::filesystem::path, resolver::ManifestRaw> manifest;
+
+  /**
+   * @brief The absolute @a std::filesystem::path to a lockfile or an inline
+   * @a flox::resolver::LockfileRaw.
+   */
+  std::optional<std::variant<std::filesystem::path, resolver::LockfileRaw>>
+    lockfile;
+
+  /**
+   * @brief The @a flox::search::SearchQuery specifying the package to
+   *        search for.
+   */
+  SearchQuery query;
+
+
+  /**
+   * @brief If `lockfile` is unset, returns `std::nullopt`.
+   *        Otherwise returns the path to the lockfile or a phony path if the
```

Comment:
Suggestion: specify the actual phony path here.

Reply:
It is irrelevant to the caller qnd should never be depended on as a part of the API.

Its UB.